include/envoy/http/filter.h

```diff
@@ -591,6 +591,7 @@ class StreamDecoderFilter : public StreamFilterBase {
    * Called with a decoded data frame.
    * @param data supplies the decoded data.
    * @param end_stream supplies whether this is the last data frame.
+   * Further note that end_stream is only true if there are no trailers.
```

Comment:
Isn't this a tautology? I.e. the fact that the data frame is last already means this.

Reply:
> Isn't this a tautology? I.e. the fact that the data frame is last already means this.

@htuch The data frame is easily mistaken for the body of the request or response. Here's the same [issue](https://github.com/envoyproxy/envoy/issues/12196).
```
virtual FilterDataStatus decodeData(Buffer::Instance& data, bool end_stream) PURE
```

Reply:
Fair, I guess this is clearer.